### PR TITLE
CP-18968: Convert install-supp-pack plugin to use new API

### PR DESCRIPTION
**Pull request by huizh**
Signed-off-by: Hui Zhang hui.zhang@citrix.com

### CODE REVIEW

File: scripts/plugins/install-supp-pack [hunk 1]
```diff
@@ -15,30 +18,56 @@ class ArgumentError(PluginError):
...
         PluginError.__init__(self, *args)
 
-class SubprocessFailure(PluginError):
-    """Raised when there was a failure attaching the VDI"""
+class InstallFailure(PluginError):
+    """Raised when there was a failure during install"""
     def __init__(self, *args):
         PluginError.__init__(self, *args)
 
+def get_from_xensource_inventory(key):
```
Comment (srowe) [inline scripts/plugins/install-supp-pack]: There's already a `readInventory` function in `xcp.environ`

Comment (huizh) [reply scripts/plugins/install-supp-pack]: Hi @srowe , thanks for the information. I've updated the commit accordingly.

**Comment by jonludlam** [comment]
Looks good.

**Comment by huizh** [comment]
Thank you, Jon.

**Pull request merged**